src/Promitor.Agents.Scraper/Scheduling/SchedulingExtensions.cs

```diff
@@ -98,74 +122,28 @@ private static void ScheduleResourceScraping(IAzureResourceDefinition resource,
-                },
-                    jobName: jobName);
-                builder.AddUnobservedTaskExceptionHandler(s =>
-                {
-                    return
-                        (_, exceptionEventArgs) =>
-                        {
-                            var exceptionLogger = s.GetService<ILogger<BackgroundJobMonitor>>();
-                            BackgroundJobMonitor.HandleException(jobName, exceptionEventArgs, exceptionLogger);
-                        };
-                });
-            });
-
-            logger.LogInformation("Scheduled scraping job {JobName} for resource collection {ResourceDiscoveryGroup} which will be reported as metric {MetricName}", jobName, resourceDiscoveryGroup.Name, metricDefinition.PrometheusMetricDefinition?.Name);
-        }
-
-        private static string GenerateResourceScrapingJobName(ScrapeDefinition<IAzureResourceDefinition> scrapeDefinition, IAzureResourceDefinition resource)
-        {
-            var jobNameBuilder = new StringBuilder();
-            jobNameBuilder.Append(scrapeDefinition.SubscriptionId);
-            jobNameBuilder.Append("-");
-            jobNameBuilder.Append(scrapeDefinition.ResourceGroupName);
-            jobNameBuilder.Append("-");
-            jobNameBuilder.Append(scrapeDefinition.PrometheusMetricDefinition.Name);
-            jobNameBuilder.Append("-");
-            jobNameBuilder.Append(resource.UniqueName);
-            jobNameBuilder.Append("-");
-            jobNameBuilder.Append(Guid.NewGuid().ToString());
-
-            return jobNameBuilder.ToString();
-        }
+                scheduleDescription = CronExpressionDescriptor.ExpressionDescriptor.GetDescription(metricsDeclaration.Metrics[0].Scraping.Schedule);
+                scheduleDescription = scheduleDescription.Replace(", ", "_").Replace(",", "_").Replace(" ", "_");
+            }
+            catch (Exception ex)
+            {
+                var metricName = metricsDeclaration.Metrics[0].AzureMetricConfiguration.MetricName;
+                logger.LogError(ex, $"Failure to create job name for scraping resources for list of metrics which includes {metricName}.");
+                scheduleDescription = $"UnparsedScheduleForMetricsIncluding{metricName}";
+            }
 
-        private static string GenerateResourceDiscoveryGroupScrapingJobName(string metricName, string resourceDiscoveryGroupName)
-        {
-            var jobNameBuilder = new StringBuilder();
-            jobNameBuilder.Append(metricName);
-            jobNameBuilder.Append("-");
-            jobNameBuilder.Append(resourceDiscoveryGroupName);
-            jobNameBuilder.Append("-");
-            jobNameBuilder.Append(Guid.NewGuid().ToString());
+            var metricsCount = metricsDeclaration.Metrics.Count;
+            var uniqueness = Guid.NewGuid().ToString("N");
 
-            return jobNameBuilder.ToString();
+            return $"Scraper-{metricsCount}Metrics-{scheduleDescription}-{uniqueness}";
```

Comment:
We don't need this I think
```suggestion
            return $"Scraper-{scheduleDescription}-{uniqueness}";
```

Reply:
It was just to keep it short but that's a nice to have indeed - Resolving